src/main/java/nus/climods/model/module/Module.java

```diff
@@ -71,6 +94,39 @@ public List<Integer> getSemesters() {
                 .map(BigDecimal::intValue).collect(Collectors.toList());
     }
 
+    /**
+     * Retrieves detailed module only if not already retrieved (lazily)
+     * @throws ApiException if error occured during API request
+     */
+    private void getDetailedModule() throws DetailedModuleRetrievalException {
+        // TODO: replace with reading of stored detailed module JSON file
+        try {
+            if (detailedModule.isEmpty()) {
+                org.openapitools.client.model.Module module = modulesApi.acadYearModulesModuleCodeJsonGet(currentAcademicYear, this.getCode());
+                detailedModule = Optional.of(module);
```

Comment:
I dont think we need a `Optional` wrapper since cmiiw the api throws a `404` when a module code is not found which our `try-catch` handles. We can probably simplify the logic without the use of `Optional`